src/universal/modules/userDashboard/containers/UserSettings/UserSettingsContainer.js

```diff
@@ -76,7 +88,56 @@ export default class UserSettingsContainer extends Component {
       router.push(nextPage);
     }
     untouch('preferredName');
-  };
+  }
+
+  uploadPicture(pictureFile) {
+    return cashay.mutate('createUserPicturePutUrl', {
+      variables: {
+        userFilename: pictureFile.name
+      }
+    })
+    .then(({data, error}) => {
+      if (error) {
+        throw new Error(error._error); // eslint-disable-line no-underscore-dangle
+      }
+      return data.createUserPicturePutUrl.picturePutUrl;
+    })
+    .then(picturePutUrl => {
+      return fetch(picturePutUrl, {
```

Comment:
Important to note. This _is_ a URL. The hash is included as a URL query param. However, remember with https URLs are encrypted (and these URLs are always https)

(N.B. http://stackoverflow.com/questions/499591/are-https-urls-encrypted)